### PR TITLE
types: buf: Fix buffer size calculation

### LINKED ISSUE #3280: tools/undump.bt broken by big buf support
Commit 868b41ca115d63bb6acd38728e1aff1ff168bb75 broke the `undump.bt` tool:
```
# build/src/bpftrace tools/undump.bt
Attaching 3 probes...
ERROR: Error loading program: kprobe:unix_stream_read_actor (try -v)
```
 cc @danobi

### PR DESCRIPTION
**Pull request by danobi**
18ca7ff5 ("semantic: Push metadata headroom calculation into type") made the mistake of thinking SizedType constructor takes size in bits. In reality it is in bytes.

This closes #3280.

##### Checklist

- [ ] Language changes are updated in `man/adoc/bpftrace.adoc`
- [x] User-visible and non-trivial changes updated in `CHANGELOG.md`
- [x] The new behaviour is covered by tests

### CODE REVIEW

**Comment by danobi** [comment]
This might warrant a backport to release branch

**Review by jordalgo — APPROVED** [review]

File: src/types.cpp [hunk 1]
```diff
@@ -435,8 +435,8 @@ SizedType CreateKSym()
 
 SizedType CreateBuffer(size_t size)
 {
-  auto metadata_headroom_bits = sizeof(AsyncEvent::Buf) * 8;
-  return SizedType(Type::buffer, size + metadata_headroom_bits);
+  auto metadata_headroom = sizeof(AsyncEvent::Buf);
```
Comment (jordalgo) [inline src/types.cpp:438]: nit, if you want to make this more explicit
```suggestion
  auto metadata_headroom_bytes = sizeof(AsyncEvent::Buf);
```

**Comment by viktormalik** [comment]
> This might warrant a backport to release branch

Yeah, probably along with #3274.

**Comment by danobi** [comment]
Actually we're good. Big buf() was merged after the release. Got lucky this time!

File: CHANGELOG.md [hunk 2]
```diff
@@ -17,6 +17,8 @@ and this project adheres to
 #### Deprecated
 #### Removed
 #### Fixed
+- Fix incorrect buf() size calculation which could lead to verifier errors
+  - [#3281](https://github.com/bpftrace/bpftrace/pull/3281)
```
[the remark this answers is not in the document]
Comment (ajor) [reply CHANGELOG.md:21]: Fixed in #3289

**Pull request merged**